src/sotauptaneclient.cc

```diff
@@ -1,17 +1,33 @@
 #include "sotauptaneclient.h"
 
 #include <unistd.h>
+
 #include <boost/make_shared.hpp>
+#include "json/json.h"
 
 #include "crypto.h"
 #include "httpclient.h"
-#include "json/json.h"
 #include "logger.h"
 #include "uptane/exceptions.h"
+#include "uptane/secondaryconfig.h"
+#include "uptane/secondaryfactory.h"
 #include "utils.h"
 
 SotaUptaneClient::SotaUptaneClient(const Config &config_in, event::Channel *events_channel_in, Uptane::Repository &repo)
-    : config(config_in), events_channel(events_channel_in), uptane_repo(repo), last_targets_version(-1) {}
+    : config(config_in), events_channel(events_channel_in), uptane_repo(repo), last_targets_version(-1) {
+  std::vector<Uptane::SecondaryConfig>::iterator it;
+  for (it = config.uptane.secondary_configs.begin(); it != config.uptane.secondary_configs.end(); ++it) {
+    std::map<std::string, boost::shared_ptr<Uptane::SecondaryInterface> >::const_iterator map_it =
```

Comment:
We can use std:set for unique elements

Reply:
The point here is to be able to quickly find a secondary by its serial number.